src/super_gradients/training/sg_model/sg_model.py

```diff
@@ -222,18 +223,21 @@ def connect_dataset_interface(self, dataset_interface: DatasetInterface, data_lo
     # FIXME - we need to resolve flake8's 'function is too complex' for this function
     def build_model(self,  # noqa: C901 - too complex
                     architecture: Union[str, nn.Module],
-                    arch_params={}, *args, **kwargs):
+                    arch_params={}, checkpoint_params={}, *args, **kwargs):
```

Comment:
@shaydeci  - Let's discuss tomorrow why are we calling load_checkpoint from within build_model... If we have successfully decoupled both of these methods, we should probably be able to call these two methods independently from SGModel...

Let's discuss this tomorrow